docs/man/amule.1
.TH AMULE 1 "September 2016" "aMule v2.3.2" "aMule"
.als B_untranslated B
.als RB_untranslated RB
.SH NAME
amule \- the all\-platform eMule p2p client
.SH SYNOPSIS
.B_untranslated amule
.RB [ \-c " " \fI<path> ]
.RB [ \-geometry " " \fI<geom> ]
.RB_untranslated [ \-o ]
.RB_untranslated [ \-r ]
.RB [ \-w " " \fI<path> ]
.RB_untranslated [ \-d ]
.RB_untranslated [ \-i ]
.RB [ \-t " " \fI<num> ]
.RI [ eD2k-link ]

.B_untranslated amule
.RB_untranslated [ \-v ]

.B_untranslated amule
.RB_untranslated [ \-h ]

.SH DESCRIPTION
.TP
\fB[ \-c\fR \fI<path>\fR, \fB\-\-config\-dir\fR=\fI<path>\fR \fB]\fR
Read config from \fI<path>\fR instead of home
.TP
\fB[ \-geometry \fI<geom>\fR \fB]\fR
Sets the geometry of the app. \fI<geom>\fR uses the same format as standard X11 apps:	[\fB=\fR][\fI<width>\fR{\fBxX\fR}\fI<height>\fR][{\fB+-\fR}\fI<xoffset>\fR{\fB+-\fR}\fI<yoffset>\fR]
.TP
.B_untranslated [ \-o\fR, \fB\-\-log\-stdout ]\fR
Prints log messages to stdout.
.TP
.B_untranslated [ \-r\fR, \fB\-\-reset\-config ]\fR
Resets config to default values.
.TP
\fB[ \-w\fR \fI<path>\fR, \fB\-\-use\-amuleweb\fR=\fI<path>\fR \fB]\fR
Specify location of amuleweb binary to \fI<path>\fR.
.TP
.B_untranslated [ \-d\fR, \fB\-\-disable\-fatal ]\fR
Don't catch fatal exceptions and don't block exit on assertions.
Useful for headless / supervised setups (systemd, watchdog scripts)
where the otherwise-modal assertion dialog prevents auto-restart.
.TP
.B_untranslated [ \-i\fR, \fB\-\-enable\-stdin ]\fR
Does not disable stdin.
.TP
\fB[ \-t\fR, \fB\-\-category\fR=\fI<num>\fR \fB]\fR
Set category for passed eD2k links to \fI<num>\fR
.TP
.B_untranslated [ \-v\fR, \fB\-\-version ]\fR
Displays the current version number.
.TP
.B_untranslated [ \-h\fR, \fB\-\-help ]\fR
Prints a short usage description.
.TP
\fB[ eD2k-link ]\fR
Adds an eD2k-link to the core.
.PP
The eD2k link to be added can be:
.RS 3
.IP \(bu 2
a file link (ed2k://|file|...), it will be added to the download queue;
.IP \(bu 2
a server link (ed2k://|server|...), it will be added to the server list;
.IP \(bu 2
a serverlist link, in which case all servers in the list will be added to the server list;
.IP \(bu 2
a magnet link.
.RE
.SH NOTES
.SS Paths
For all options which take a \fI<path>\fR value, if the \fIpath\fR contains no
directory part (i.e. just a plain filename), then it is considered to be under
the aMule configuration directory, \fI~/.aMule\fR.
.SH FILES
~/.aMule/*
.SH REPORTING BUGS
Please report bugs either on our forum (\fIhttps://github.com/amule-org/amule/discussions\fR), or in our bugtracker (\fIhttps://github.com/amule-org/amule/issues\fR).
Please do not report bugs in e-mail, neither to our mailing list nor directly to any team member.
.SH COPYRIGHT
aMule and all of its related utilities are distributed under the GNU General Public License.
.SH SEE ALSO
.B_untranslated alc\fR(1), \fBalcc\fR(1), \fBamuled\fR(1), \fBamulecmd\fR(1), \fBamulegui\fR(1), \fBamuleweb\fR(1), \fBcas\fR(1), \fBed2k\fR(1), \fBwxcas\fR(1), \fBxas\fR(1)
.SH AUTHOR
This manpage was written by Vollstrecker <amule@vollstreckernet.de>


ChangeLog
Version 2.3.4 - The "can we have two releases in the same year?" version.
----------
2021-??-??


--------------------------------------------------------------------------------

Version 2.3.3 - The "delayed too many times, but still not too late" version.
----------
2021-02-07

	alesnav:
		* Permit amuleweb in an iframe

	Dmitry Smirnov:
		* Added mimetype-hanlder to amule.desktop
		  originally in debian

	gaffatape:
		* Workaround for bug in wxWidgets causing aMule to crash on
		  closing the last search tab

	GonoszTopi:
		* Fix restoring toolbar orientation on 'Cancel'
		* Features ported from eMule:
			- Find best keyword for Kad search (0.49c)
		* Various fixes to gdlib detection
		* Support for compiling aMule with wxGTK under Cygwin
		* Various small fixes to the MacOSX packager script
		* Workaround for wxWidgets bug #17740 (Wrong path used to
		  search for translations), fixes GitHub issue #74 (.mo files
		  for translation are not loaded if amule is built to install
		  to a prefix other than /usr (e.g. /usr/local).)
		* Portability fixes
		* Handle the deprecation of std::auto_ptr in C++11 by using
		  std::unique_ptr instead.
		* Fix linking with readline if configure uses caching
		* Remove 'web search' search type as there are no search
		  engines to support
		* Readline related fixes and improvements in amulecmd:
			- Set the proper application name (amulecmd) for
			  readline. This allows conditional parsing of the
			  readline configuration files (like ~/.inputrc). See
			  the GNU Readline Library documentation (section 1.3.2)
			  for more details.
			- Set up command completion on pressing the TAB key.
			  Since command names are case insensitive, completion
			  uses the default capitalization (shown by the 'Help'
			  command). Completion also works for the
			  'Help <command name>' construct.
			- Pressing <Enter> on an empty line won't give 'Syntax
			  error!' anymore, the empty line is silently ignored.
		* Flex and bison generated files will no longer be regenerated
		  occasionally on the user's system.
		* Proxy authorization fixes
		* Explicitly disable support for autoconf >= 2.70
		* Fix compilation using C++17

	jEsuSdA:
		* Added papirus theme

	KeyofBlueS:
		* Fix compilation with newer bfd

	kinote:
		* Fix integer conversions in amulecmd

	persmule:
		* Fix sharing directories with non-ASCII names with aMule daemon
		* Fix potential invalid detection for empty wxCharBuffer

	Phoenix:
		* Fixed libupnp API breakage.

	poplite:
		* Fixed issue #99 by properly initializing wxSockets in the
		  remote gui

	Sandro Tosi:
		* remove garbage from gdlib
		  originally in debian

	sc0w:
		* Fixed compilation with upnp-1.8
		* C++11 fixes
		* Only handle mouse-up events if the mouse-down event happened
		  on the same tab
		* Fix issue #114: Switching search result tabs doesn't work
		* Fix a number of build warnings
		* Fix a few memory leaks
		* Various small code improvements
		* Add option to show notifications when finished downloading
		* Fixed issue #54: the minimize button doesn't work propertly
		  when "minimize to tray icon" is activated on preferences
		* Fixed issue #59: start minimized doesn't work
		* Fixed issue #225: amule shows wrong 100% when files aren't
		  downloaded
		* Show speeds in MB/s instead of kB/s when appropriate

	Sebastiano Pistore:
		* Unified translation file headers to make it easier to identify
		  contributors

	Thibaut Lassalle:
		* Add README.md

	Tom Li:
		* OScopeCtrl.cpp: use Refresh() instead of ProcessEvent() to
		  refresh the UI, fix issue #146.

	Tommy Jerry Mairo:
		* Fixed issue #119: API mismatch with Crypto++ 6.0.0

	Vollstrecker:
		* Initial support for detecting pkg-config based gdlib versions
		* Compilation fixes for VS2015
		* Fixed compilation with wx-3.1
		* Added cmake-support (Linux and Windows tested)
		* Dropped support for cryptopp versions before 5.5

	xinhuang:
		* Fix crash when shared files changed on Shared files tab

--------------------------------------------------------------------------------

Version 2.3.2 - The "don't close tabs" version.
----------
2016-09-16

	circulosmeos:
		* amuleweb cookies read behind a "lowercase" proxy (http/2 compliant)

	Dan64:
		* Read AICH root hashes from binary eMuleCollection files

	gnazio:
		* Fixed configure of libpng when sh != bash

	GonoszTopi:
		* Allow any kind of file links in collections
		* Make UPnP log go to the appropriate category (#1681)
		* Use the C++ compiler to check for features we're about to use
		  in C++ source files (#1572)
		* Adapt configure to handle subversion repository clones in
		  mercurial and git
		* Fix bfd detection for systems missing -liberty
		* Add support for 'canceled.met' to the fileview tool
		* Implemented the missing 'show shared' command in amulecmd
		* Fix check for a valid locale config entry being done *after*
		  the locale has been set
		* Fix Crypto++ detection routine happily using build library
		  for host when cross-compiling
		* Fix bug #1711: Kad Info tab should display my own user ID
		* Fix wrong sized images in the priscilla skin
		* Fix bug #1423: Add the possibility to not create sparse files
		* Fix bug #1743: crash when importing part files
		* Review assertions in the eD2k and Kademlia code and remove
		  those that could be triggered by a malformed incoming packet
		* Fix a possible uninitialized variable access and a definite
		  memory leak in the UPnP code

	iz0bbz:
		* Fix WinSock library version detection with MinGW-w64
		* Fix compilation with MinGW-w64

	kam:
		* WebServer serves .json, .manifest and .appacache files

	Mr Hyde:
		* Fix "Disable computer's timed standby mode" for MacOS 10.9 Mavericks
		* Compilation fix on Linux Slackware 64bit and MacOS 10.9.4
		* Fix configure on Mac

	sc0w:
		* Add ability to use middle-click to close search tabs

	Stu Redman:
		* Display hashing progress in progress bars

docker/x86/amule.sh
#!/usr/bin/env bash

# Uncomment for debug
#set -x

AMULE_UID=${PUID:-5000}
AMULE_GID=${PGID:-5000}
AMULE_WEBUI_TEMPLATE=${WEBUI_TEMPLATE:-default}

AMULE_HOME=/home/amule/.aMule
AMULE_INCOMING=/incoming
AMULE_TEMP=/temp
AMULE_CONF=${AMULE_HOME}/amule.conf
REMOTE_CONF=${AMULE_HOME}/remote.conf

AMULE_GROUP="amule"
if grep -q ":${AMULE_GID}:" /etc/group; then
    echo "Group ${AMULE_GID} already exists. Won't be created."
    AMULE_GROUP=$(getent group "${AMULE_GID}" | cut -d: -f1)
    echo "Group ${AMULE_GROUP} with GID ${AMULE_GID} will be used as amule group."
else
    addgroup -g "${AMULE_GID}" amule
fi

if grep -q ":${AMULE_UID}:" /etc/passwd; then
    echo "User ${AMULE_UID} already exists. Won't be added."
else
    adduser -S -s /sbin/nologin -u "${AMULE_UID}" -h "/home/amule" -G "${AMULE_GROUP}" amule
fi
    
if [[ ! -d "${AMULE_INCOMING}" ]]; then
    echo "Directory ${AMULE_INCOMING} does not exists. Creating ..."
    mkdir -p ${AMULE_INCOMING}
    chown -R "${AMULE_UID}:${AMULE_GID}" "${AMULE_INCOMING}"
fi

if [[ ! -d "${AMULE_TEMP}" ]]; then
    echo "Directory ${AMULE_TEMP} does not exists. Creating ..."
    mkdir -p ${AMULE_TEMP}
    chown -R "${AMULE_UID}:${AMULE_GID}" "${AMULE_TEMP}"
fi

if [[ -z "${GUI_PWD}" ]]; then
    AMULE_GUI_PWD=$(pwgen -s 64)
    echo "No GUI password specified, using generated one: ${AMULE_GUI_PWD}"
else
    AMULE_GUI_PWD="${GUI_PWD}"
fi
AMULE_GUI_ENCODED_PWD=$(echo -n "${AMULE_GUI_PWD}" | md5sum | cut -d ' ' -f 1)

if [[ -z "${WEBUI_PWD}" ]]; then
    AMULE_WEBUI_PWD=$(pwgen -s 64)
    echo "No web UI password specified, using generated one: ${AMULE_WEBUI_PWD}"
else
    AMULE_WEBUI_PWD="${WEBUI_PWD}"
fi
AMULE_WEBUI_ENCODED_PWD=$(echo -n "${AMULE_WEBUI_PWD}" | md5sum | cut -d ' ' -f 1)

if [[ ! -d ${AMULE_HOME} ]]; then
    echo "${AMULE_HOME} directory NOT found. Creating directory ..."
    sudo -H -u '#'"${AMULE_UID}" sh -c "mkdir -p ${AMULE_HOME}"
fi

if [[ ! -f ${AMULE_CONF} ]]; then
    echo "${AMULE_CONF} file NOT found. Generating new default configuration ..."
    cat > ${AMULE_CONF} <<- EOM
[eMule]
AppVersion=2.3.1
Nick=http://www.aMule.org
QueueSizePref=50
MaxUpload=0
MaxDownload=0
SlotAllocation=2
Port=4662
UDPPort=4672
UDPEnable=1
Address=
Autoconnect=1
MaxSourcesPerFile=300
MaxConnections=500
MaxConnectionsPerFiveSeconds=20
RemoveDeadServer=1
DeadServerRetry=3
ServerKeepAliveTimeout=0
Reconnect=1
Scoresystem=1
Serverlist=0
AddServerListFromServer=0
AddServerListFromClient=0
SafeServerConnect=0
AutoConnectStaticOnly=0
UPnPEnabled=0
UPnPTCPPort=50000
SmartIdCheck=1
ConnectToKad=1
ConnectToED2K=1
TempDir=${AMULE_TEMP}
IncomingDir=${AMULE_INCOMING}
ICH=1
AICHTrust=0
CheckDiskspace=1
MinFreeDiskSpace=1
AddNewFilesPaused=0
PreviewPrio=0
ManualHighPrio=0
StartNextFile=0
StartNextFileSameCat=0
StartNextFileAlpha=0
FileBufferSizePref=16
DAPPref=1
UAPPref=1
AllocateFullFile=0
OSDirectory=${AMULE_HOME}
OnlineSignature=0
OnlineSignatureUpdate=5
EnableTrayIcon=0
MinToTray=0
ConfirmExit=1
StartupMinimized=0
3DDepth=10
ToolTipDelay=1
ShowOverhead=0
ShowInfoOnCatTabs=1
VerticalToolbar=0
GeoIPEnabled=1
ShowVersionOnTitle=0
VideoPlayer=
StatGraphsInterval=3
statsInterval=30
DownloadCapacity=300
UploadCapacity=100
StatsAverageMinutes=5
VariousStatisticsMaxValue=100
SeeShare=2
FilterLanIPs=1
ParanoidFiltering=1
IPFilterAutoLoad=1
IPFilterURL=
FilterLevel=127
IPFilterSystem=0
FilterMessages=1
FilterAllMessages=0
MessagesFromFriendsOnly=0
MessageFromValidSourcesOnly=1
FilterWordMessages=0
MessageFilter=
ShowMessagesInLog=1
FilterComments=0
CommentFilter=
ShareHiddenFiles=0
AutoSortDownloads=0
NewVersionCheck=0
AdvancedSpamFilter=1
MessageUseCaptchas=1
Language=
SplitterbarPosition=75
YourHostname=
DateTimeFormat=%A, %x, %X
AllcatType=0
ShowAllNotCats=0
SmartIdState=0
DropSlowSources=0
KadNodesUrl=http://upd.emule-security.org/nodes.dat
Ed2kServersUrl=http://gruk.org/server.met.gz
ShowRatesOnTitle=0
GeoLiteCountryUpdateUrl=http://geolite.maxmind.com/download/geoip/database/GeoLiteCountry/GeoIP.dat.gz
StatsServerName=Shorty's ED2K stats
StatsServerURL=http://ed2k.shortypower.dyndns.org/?hash=
[Browser]
OpenPageInTab=1
CustomBrowserString=
[Proxy]
ProxyEnableProxy=0
ProxyType=0
ProxyName=
ProxyPort=1080
ProxyEnablePassword=0
ProxyUser=
ProxyPassword=
[ExternalConnect]
UseSrcSeeds=0
AcceptExternalConnections=1
ECAddress=
ECPort=4712
ECPassword=${AMULE_GUI_ENCODED_PWD}
UPnPECEnabled=0
ShowProgressBar=1
ShowPercent=1
UseSecIdent=1
IpFilterClients=1
IpFilterServers=1
TransmitOnlyUploadingClients=0
[WebServer]
Enabled=1
Password=${AMULE_WEBUI_ENCODED_PWD}
PasswordLow=
Port=4711
WebUPnPTCPPort=50001
UPnPWebServerEnabled=0
UseGzip=1
UseLowRightsUser=0
PageRefreshTime=120
Template=${AMULE_WEBUI_TEMPLATE}
Path=amuleweb
[GUI]
HideOnClose=0
[Razor_Preferences]
FastED2KLinksHandler=1
[SkinGUIOptions]
Skin=
[Statistics]
MaxClientVersions=0
[Obfuscation]
IsClientCryptLayerSupported=1
IsCryptLayerRequested=1
IsClientCryptLayerRequired=0
CryptoPaddingLenght=254
CryptoKadUDPKey=338883508
[PowerManagement]
PreventSleepWhileDownloading=0
[UserEvents]
[UserEvents/DownloadCompleted]
CoreEnabled=0
CoreCommand=
GUIEnabled=0
GUICommand=
[UserEvents/NewChatSession]
CoreEnabled=0
CoreCommand=
GUIEnabled=0
GUICommand=
[UserEvents/OutOfDiskSpace]
CoreEnabled=0
CoreCommand=
GUIEnabled=0
GUICommand=
[UserEvents/ErrorOnCompletion]
CoreEnabled=0
CoreCommand=
GUIEnabled=0
GUICommand=
[HTTPDownload]
URL_1=
EOM
    echo "${AMULE_CONF} successfullly generated."
else
    echo "${AMULE_CONF} file found. Using existing configuration."
fi

if [[ ! -f ${REMOTE_CONF} ]]; then
    echo "${REMOTE_CONF} file NOT found. Generating new default configuration ..."
    cat > ${REMOTE_CONF} <<- EOM
Locale=
[EC]
Host=localhost
Port=4712
Password=${AMULE_GUI_ENCODED_PWD}
[Webserver]
Port=4711
UPnPWebServerEnabled=0
UPnPTCPPort=50001
Template=${AMULE_WEBUI_TEMPLATE}
UseGzip=1
AllowGuest=0
AdminPassword=${AMULE_WEBUI_ENCODED_PWD}
GuestPassword=
EOM
    echo "${REMOTE_CONF} successfullly generated."
else
    echo "${REMOTE_CONF} file found. Using existing configuration."
fi

# Define web UI theme if needed
if [[ "${AMULE_WEBUI_TEMPLATE}" = "default" ]]
then
    echo "Using default theme for web UI."
else
    sed -i 's,^\(Template[ ]*=\).*,\1'"${AMULE_WEBUI_TEMPLATE}"',g' ${REMOTE_CONF}
    sed -i 's,^\(Template[ ]*=\).*,\1'"${AMULE_WEBUI_TEMPLATE}"',g' ${AMULE_CONF}
    echo "Switched web UI theme to ${AMULE_WEBUI_TEMPLATE}"
fi

chown -R "${AMULE_UID}:${AMULE_GID}" /home/amule

sudo -H -u '#'"${AMULE_UID}" sh -c "amuled -c ${AMULE_HOME} -o"
